packages/edit-site/src/components/left-sidebar/navigation-panel/template-navigation-items.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	__experimentalNavigationItem as NavigationItem,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import TemplatePreview from './template-preview';
+import { NavigationPanelPreviewFill } from '.';
+import { ITEM_CONTENTS } from './constants';
+
+export default function TemplateNavigationItems( {
+	entityType = 'wp_template',
+	templates,
+	onActivate,
+} ) {
+	const [ hoveredTemplate, setHoveredTemplate ] = useState();
+
+	const onMouseEnterTemplate = ( template ) => setHoveredTemplate( template );
+	const onMouseLeaveTemplate = () => setHoveredTemplate( null );
+
+	if ( ! templates ) {
+		return null;
+	}
+
+	if ( ! Array.isArray( templates ) ) {
+		templates = [ templates ];
+	}
+
+	const itemPrefix =
+		entityType === 'wp_template_part' ? 'template-part' : 'template';
+
+	return (
+		<>
+			{ templates.map( ( template ) => {
+				const { title, description } =
+					ITEM_CONTENTS[ template.slug ] ?? {};
+				const key = `${ itemPrefix }-${ template.id }`;
+
+				return (
+					<NavigationItem
+						className="edit-site-navigation-panel__template-item"
+						key={ key }
+						item={ key }
+						title={ title || template.slug }
+					>
+						<Button
+							onClick={ () => onActivate( template.id ) }
+							onMouseEnter={ () =>
+								onMouseEnterTemplate( template )
+							}
+							onMouseLeave={ onMouseLeaveTemplate }
+						>
+							{ title || template.slug }
```

Comment:
Wonder if it makes more sense to invert this? 🤔 
`{ template.slug || title }`

Reply:
Slug isn't an optional field, so our "default titles" would never be shown. I don't think 'index' (slug) would be more descriptive than "Default (index)".

Reply:
Maybe you meant `template.title.rendered`? That would make sense :D In this case we need to make sure it is different from slug.

Reply:
I was more thinking around the presence of an autodraft (and therefore a slug) or falling back to the original, default template.
Thinking about it, I'm pretty sure I got confused by the whole thing though 😄

Reply:
We always have an auto-draft now. They are created automatically. Though we would have `slug` even if we didn't have auto-drafts.